projects/angular-calendar/src/modules/common/calendar-a11y.provider.ts

```diff
@@ -0,0 +1,162 @@
+import { Injectable } from '@angular/core';
+import { formatDate, I18nPluralPipe } from '@angular/common';
+import { A11yParams } from './calendar-a11y.interface';
+
+/**
+ * This class is responsible for adding accessibility to the calendar.
+ * You may override any of its methods via angulars DI to suit your requirements.
+ * For example:
+ *
+ * ```typescript
+ * import { A11yParams, CalendarA11y } from 'angular-calendar';
+ * import { formatDate } from '@angular/common';
+ *
+ * // adding your own a11y params
+ * export interface CustomA11yParams extends A11yParams {
+ *   isDrSuess?: boolean;
+ * }
+ *
+ * export class CustomCalendarA11y extends CalendarA11y {
+ *     constructor(public i18nPlural: I18nPluralPipe) {
+ *       super(i18nPlural);
+ *     }
+ *
+ *   // overriding a function
+ *   public openDayEventsLandmark({ date, locale, isDrSuess }: CustomA11yParams): string {
+ *     if (isDrSuess) {
+ *       return `
+ *         ${formatDate(date, 'EEEE MMMM d', locale)}
+ *          Today you are you! That is truer than true! There is no one alive
+ *          who is you-er than you!
+ *       `;
+ *     }
+ *   }
+ * }
+ *
+ * // in your component that uses the calendar
+ * providers: [{
+ *  provide: CalendarA11y,
+ *  useClass: CustomCalendarA11y
+ * }]
+ * ```
+ */
+@Injectable()
+export class CalendarA11y {
+  /**
+   * Pluralization for events
+   */
+  eventPluralMap: { [k: string]: string } = {
+    '=0': 'No events',
+    '=1': 'One event',
+    other: '# events'
+  };
+
+  constructor(public i18nPlural: I18nPluralPipe) {}
```

Comment:
Let's make this `protected`, so that it's still usable within child classes but not usable from outside the class